### PR TITLE
Draft caching update

### PR DESCRIPTION
**Pull request by ethangj**
WIP

### CODE REVIEW

File: _pro/builds-and-configuration/caching.md [hunk 1]
```diff
@@ -28,9 +28,9 @@ This tutorial describes the way caching works on Codeship's infrastructure durin
 
 ## Using Caching
 
-As a way to speed up your build pipelines, Codeship supports persistent caching. This means we'll export your final image as a compressed tar file, which is stored in an encrypted bucket on AWS S3.
+As a way to speed up your build pipelines, Codeship supports image caching that works per-service and per-branch.
 
-For future builds, we'll import the image to your build's machine and repopulate the local Docker image cache. Then, when the service image is rebuilt, only the layers that changed will need to be updated. [Read more about optimizing your Dockerfile for caching.](#optimizing-your-build-to-use-the-docker-image-cache)
+This mean's at the end of builds, we will push any images you enabling caching for out to a secure registry and pull the image at the start of future builds to reuse as many of the image layers as possible rather than rebuilding them. [Read more about optimizing your Dockerfile for caching.](#optimizing-your-build-to-use-the-docker-image-cache)
```
Comment (mlocher) [inline _pro/builds-and-configuration/caching.md]: Should we modify that sentence to read _after your build finished..._ to make it even more clear, that the cache does not count against the build times?

File: _pro/builds-and-configuration/services.md [hunk 2]
```diff
@@ -195,7 +195,7 @@ The boolean directive `add_docker` is available. If specified for a service, it
 See [add_docker](https://github.com/codeship/codeship-tool-examples/tree/master/14.add_docker) for an example using [Docker-in-Docker](https://registry.hub.docker.com/u/jpetazzo/dind).
 
 ### Caching the Docker image
-Caching is declared per service. For a service with caching enabled, Codeship will store your image remotely in an encrypted S3 bucket, and then use that image to repopulate the local build cache on future build runs. This prevents the Docker image from building from scratch each time, to save time and speed up your CI/CD process. By default, we will fall back to the latest image that was built on the `master` branch.
+Caching is declared per service. For a service with caching enabled Codeship will push your image out to a secure image registry, and then use pull and use that image on future builds. This prevents the Docker image from building from scratch each time, to save time and speed up your CI/CD process. By default, we will fall back to the latest image that was built on the `master` branch.
```
Comment (mlocher) [inline _pro/builds-and-configuration/services.md]: > ...and then use pull and use that image on future builds.

I think we can clean up this part of the sentence a bit.

**Comment by ethangj** [comment]
@mlocher could you re-review?

File: _pro/builds-and-configuration/caching.md [hunk 3]
```diff
@@ -71,6 +71,8 @@ app:
 
 If the `default_cache_branch` directive is not present, we will always use `master` as your cache fallback branch by default.
 
+Note that branch-cache images are automatically removed after 90 days of inactivity.
```
Comment (mlocher) [inline _pro/builds-and-configuration/caching.md]: Would move that to an info block.

**Pull request merged**